client/src/main/scala/drt/client/components/MonthlyShiftsUtil.scala

```diff
@@ -0,0 +1,139 @@
+      case "daily" => viewingDate
+      case _ => SDate.firstDayOfMonth(viewingDate)
+    }
+  }
+
+  def iteratorForShiftAssignment(isShiftEndAfterMidNight: Boolean,
+                                 day: Int,
+                                 start: SDateLike,
+                                 end: SDateLike,
+                                 endHour: Int,
+                                 endMinute: Int,
+                                 isFirstDayForShiftEndAfterMidNight: Boolean,
+                                 addToIndex: Int,
+                                 interval: Int,
+                                 terminal: Terminal, s: Shift,
+                                 shifts: ShiftAssignments) = {
+    val dayAssignments: Seq[StaffAssignmentLike] = shifts.assignments.filter(assignment => assignment.start >= start.millisSinceEpoch && assignment.end <= end.millisSinceEpoch)
+    Iterator.iterate(start) { intervalTime =>
+      if ((intervalTime.getMinutes == 30 && interval == 60) || (intervalTime.getHours == endHour && interval == 60 && endMinute == 30)) {
+        intervalTime.addMinutes(30)
+      } else {
+        intervalTime.addMinutes(interval)
+      }
+    }.takeWhile(_ < end).toSeq.zipWithIndex.map { case (currentTime, index) =>
+      val nextTime = if ((currentTime.getMinutes == 30 && interval == 60) || (currentTime.getHours == endHour && interval == 60 && endMinute == 30)) {
+        currentTime.addMinutes(30)
+      } else {
+        currentTime.addMinutes(interval)
+      }
+      val matchedAssignments = dayAssignments.find(assignment => assignment.start == currentTime.millisSinceEpoch && assignment.terminal == terminal)
+      matchedAssignments match {
+        case Some(sa) =>
+          StaffTableEntry(
+            column = day,
+            row = if (isShiftEndAfterMidNight) index + addToIndex else index,
+            name = sa.name,
+            staffNumber = sa.numberOfStaff,
+            startTime = ShiftDate(currentTime.getFullYear, currentTime.getMonth, currentTime.getDate, currentTime.getHours, currentTime.getMinutes),
+            endTime = ShiftDate(nextTime.getFullYear, nextTime.getMonth, nextTime.getDate, nextTime.getHours, nextTime.getMinutes)
+          )
+        case None =>
+          StaffTableEntry(
+            column = day,
+            row = if (isShiftEndAfterMidNight) index + addToIndex else index,
+            name = s.shiftName,
+            staffNumber = if (isFirstDayForShiftEndAfterMidNight) 0 else s.staffNumber,
+            startTime = ShiftDate(currentTime.getFullYear, currentTime.getMonth, currentTime.getDate, currentTime.getHours, currentTime.getMinutes),
+            endTime = ShiftDate(nextTime.getFullYear, nextTime.getMonth, nextTime.getDate, nextTime.getHours, nextTime.getMinutes)
+          )
+      }
+    }
+  }
+
+  def assignmentsForShift(firstDay: SDateLike, daysCount: Int, interval: Int, terminal: Terminal, s: Shift, shifts: ShiftAssignments): Seq[StaffTableEntry] = {
```

Comment:
`s` is a `Shift`, and `shifts` is `ShiftAssignments`. Can we rename to make it clear?

And the function could maybe be called `staffTableEntriesForShift`